common4j/src/main/com/microsoft/identity/common/java/cache/AbstractAccountCredentialCache.java

```diff
@@ -276,6 +277,36 @@ protected List<Credential> getCredentialsFilteredByInternal(@NonNull final List<
         return matchingCredentials;
     }
 
+    /**
+     * The application identifier field for access token records previously included a SHA-1 signing certificate hash.
+     * This method identifies such tokens, as the application identifier should now contain a SHA-512 signing certificate hash.
+     * @param credential token record.
+     * @return true if an access token containing a SHA-1 app identifier hash; false otherwise.
+     */
+    public static boolean isSha1ApplicationIdentifierAccessToken(@NonNull final Credential credential) {
+        final String methodTag = TAG + ":isSha1ApplicationIdentifierAccessToken";
+        if (credential instanceof AccessTokenRecord) {
+            final AccessTokenRecord accessToken = (AccessTokenRecord) credential;
+            final String tokenAppIdentifier = accessToken.getApplicationIdentifier();
+            if (tokenAppIdentifier != null
+                    && applicationIdentifierContainsSha1(tokenAppIdentifier)) {
+                Logger.info(methodTag, "Identified old access token with app identifier containing SHA-1. This token shall be removed, and a new access token should be re-acquired with a SHA-512 app identifier.");
```

Comment:
>Logger.info(methodTag, "Identified old access token with app identifier containing SHA-1. This token shall be removed, and a new access token should be re-acquired with a SHA-512 app identifier."); [](http://example.com/codeflow?start=16&length=197)

This log line might be better one line further up, where the decision is actually made to do the delete.  It probably doesn't matter much, but this function is really just about determining if something is sha1, while the caller is the one that decides sha1 == delete